Fix cg_configure Fortran/C interface type mismatches

brtnfld opened the pull request:
Use INTEGER(C_INT) in Fortran parameter and int in C for 'what' parameter. 
Fix value type handling for int vs size_t config options.

Fixes #875
<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Fix type mismatches in `cg_configure` Fortran/C interface and update value type handling.
> 
>   - **Type Mismatch Fixes**:
>     - Change `what` parameter type to `int` in `cg_configure_c_ptr()` and `cg_configure_c_funptr()` in `cg_ftoc.c`.
>     - Change `what` parameter type to `INTEGER(C_INT)` in `cg_configure_ptr()` and `cg_configure_funptr()` in `cgns_f.F90`.
>   - **Value Type Handling**:
>     - Fix handling of `int` vs `size_t` for configuration options in `cg_configure_c_ptr()` in `cg_ftoc.c`.
>     - Handle `size_t` values by casting to pointer and `int` values by casting to `size_t` then to pointer.
>   - **Constants Update**:
>     - Update Fortran constants to `INTEGER(C_INT)` in `cgns_f.F90`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=CGNS%2FCGNS&utm_source=github&utm_medium=referral)<sup> for 9cd6880868c6183fcafe8c94cae90e71523e06b1. You can [customize](https://app.ellipsis.dev/CGNS/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

brtnfld committed:
Fix cg_configure Fortran/C interface type mismatches

Use INTEGER(C_INT) in Fortran paramerter and int in C for 'what' parameter.
Fix value type handling for int vs size_t config options.

Fixes #875

brtnfld committed:
Use intptr_t for integer-to-pointer conversions

Improves portability when casting numeric values to void* in
cg_configure_c_ptr.

brtnfld commented on src/cg_ftoc.c:
```diff
@@ -119,47 +119,61 @@ static void string_2_F_string(char *c_string, char *string,
+             *what == CG_CONFIG_HDF5_BUFFER ||
+             *what == CG_CONFIG_HDF5_SIEVE_BUF_SIZE ||
+             *what == CG_CONFIG_HDF5_ELINK_CACHE_SIZE ) {
+    /* For these options, the value is a size_t that needs to be passed by value (as pointer) */
```
updated with suggestion

MicK7 commented:
I am fine with the `what` modification.
Don't we need to modify `ADFH_configure` too, to handle values that are not `size_t`, I am a bit confused ?
The fortran test will also need modification.
When modifying the C part we could completely remove the cgint_f references and let the compiler handle conversion with ISO bindings.

brtnfld committed:
Fix ADFH_Configure size_t value handling

Add explicit (size_t) casts for configuration options that receive
size_t values encoded as pointers.

brtnfld commented:
I updated ADFH_configure, good catch!
I don't see any changes needed for the Fortran tests. Did you see a particular change? I did miss some configuration parameter tests, like the metadata options, though.

brtnfld committed:
Addressed Codacy issue, removed redundant #include <stdint.h>

brtnfld committed:
Fix CodeQL warnings for size_t value assignments

brtnfld committed:
removed the size_t cast

brtnfld commented:
@MicK7, would you prefer this PR or the other one? I'm finished with this one.

MicK7 commented:
Let's go with this one. The change is minor in this one thus easy to apply as a patch for linux distribution.
